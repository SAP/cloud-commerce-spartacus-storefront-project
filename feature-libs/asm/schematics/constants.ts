export const ASM_MODULE = 'AsmModule';
export const ASM_FEATURE_NAME = 'asm';
export const ASM_ROOT_MODULE = 'AsmRootModule';
export const SPARTACUS_ASM = '@spartacus/asm';
export const SPARTACUS_ASM_ROOT = `${SPARTACUS_ASM}/root`;
export const SPARTACUS_ASM_ASSETS = `${SPARTACUS_ASM}/assets`;
export const ASM_TRANSLATIONS = 'asmTranslations';
export const ASM_TRANSLATION_CHUNKS_CONFIG = 'asmTranslationChunksConfig';

export const CLI_ASM_FEATURE = 'asm';
