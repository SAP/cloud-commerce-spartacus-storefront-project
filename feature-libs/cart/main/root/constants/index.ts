export * from './order-entires.context';
