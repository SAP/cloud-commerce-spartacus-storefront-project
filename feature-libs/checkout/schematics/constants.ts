export const CHECKOUT_FOLDER_NAME = 'checkout';
export const SCSS_FILE_NAME = 'checkout.scss';

export const CHECKOUT_BASE_FEATURE_NAME_CONSTANT = 'CHECKOUT_FEATURE';
export const CHECKOUT_BASE_MODULE_NAME = 'Checkout';
export const CHECKOUT_BASE_MODULE = 'CheckoutModule';
export const CHECKOUT_BASE_ROOT_MODULE = 'CheckoutRootModule';
export const CHECKOUT_BASE_TRANSLATIONS = 'checkoutTranslations';
export const CHECKOUT_BASE_TRANSLATION_CHUNKS_CONFIG =
  'checkoutTranslationChunksConfig';

export const CHECKOUT_B2B_MODULE = 'CheckoutB2BModule';
export const CHECKOUT_B2B_ROOT_MODULE = 'CheckoutB2BRootModule';
export const CHECKOUT_B2B_TRANSLATIONS = 'checkoutB2BTranslations';
export const CHECKOUT_B2B_TRANSLATION_CHUNKS_CONFIG =
  'checkoutB2BTranslationChunksConfig';

export const CHECKOUT_SCHEDULED_REPLENISHMENT_MODULE =
  'CheckoutScheduledReplenishmentModule';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_ROOT_MODULE =
  'CheckoutScheduledReplenishmentRootModule';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_TRANSLATIONS =
  'checkoutScheduledReplenishmentTranslations';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_TRANSLATION_CHUNKS_CONFIG =
  'checkoutScheduledReplenishmentTranslationChunksConfig';
