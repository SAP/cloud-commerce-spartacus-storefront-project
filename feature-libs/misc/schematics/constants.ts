export const STOREFINDER_SCSS_FILE_NAME = 'storefinder.scss';
