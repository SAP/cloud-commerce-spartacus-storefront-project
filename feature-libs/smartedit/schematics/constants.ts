export const SMARTEDIT_MODULE = 'SmartEditModule';
export const SMARTEDIT_FEATURE_NAME = 'smartEdit';
export const SMARTEDIT_ROOT_MODULE = 'SmartEditRootModule';
export const SPARTACUS_SMARTEDIT = '@spartacus/smartedit';
export const SPARTACUS_SMARTEDIT_ROOT = `${SPARTACUS_SMARTEDIT}/root`;
export const SPARTACUS_SMARTEDIT_ASSETS = 'smartedit/assets';

export const CLI_SMARTEDIT_FEATURE = 'SmartEdit';
