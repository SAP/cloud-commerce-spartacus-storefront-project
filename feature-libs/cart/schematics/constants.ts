import { SPARTACUS_CART } from '@spartacus/schematics';

export const CART_FOLDER_NAME = 'cart';
export const SCSS_FILE_NAME = 'cart.scss';

export const CART_SAVED_CART_MODULE_NAME = 'CartSavedCart';
export const CART_SAVED_CART_FEATURE_NAME_CONSTANT = 'CART_SAVED_CART_FEATURE';
export const SAVED_CART_MODULE = 'SavedCartModule';
export const SAVED_CART_ROOT_MODULE = 'SavedCartRootModule';
export const SPARTACUS_SAVED_CART = `${SPARTACUS_CART}/saved-cart`;
export const SPARTACUS_SAVED_CART_ROOT = `${SPARTACUS_SAVED_CART}/root`;
export const SPARTACUS_SAVED_CART_ASSETS = `${SPARTACUS_SAVED_CART}/assets`;
export const SAVED_CART_TRANSLATIONS = 'savedCartTranslations';
export const SAVED_CART_TRANSLATION_CHUNKS_CONFIG =
  'savedCartTranslationChunksConfig';
