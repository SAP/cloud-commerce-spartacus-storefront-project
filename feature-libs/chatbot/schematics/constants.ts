import { SPARTACUS_CHATBOT } from '@spartacus/schematics';

export const CHATBOT_FOLDER_NAME = 'chatbot';
export const CHATBOT_MODULE_NAME = 'ChatBot';

export const CHATBOT_FEATURE_NAME_CONSTANT = 'CHAT_BOT_FEATURE';
export const CHATBOT_ROOT_MODULE = 'ChatBotRootModule';
export const SPARTACUS_CHATBOT_ROOT = `${SPARTACUS_CHATBOT}/root`;
export const SPARTACUS_CHATBOT_ASSETS = `${SPARTACUS_CHATBOT}/assets`;
export const CHATBOT_TRANSLATIONS = 'chatBotTranslations';
export const CHATBOT_TRANSLATION_CHUNKS_CONFIG =
  'chatBotTranslationChunksConfig';
