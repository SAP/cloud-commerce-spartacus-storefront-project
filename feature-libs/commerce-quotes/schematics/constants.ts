import { SPARTACUS_COMMERCE_QUOTES } from '@spartacus/schematics';

export const COMMERCE_QUOTES_FOLDER_NAME = 'commerce-quotes';
export const COMMERCE_QUOTES_MODULE_NAME = 'CommerceQuotes';

export const COMMERCE_QUOTES_FEATURE_NAME_CONSTANT = 'COMMERCE_QUOTES_FEATURE';
export const SPARTACUS_COMMERCE_QUOTES_ROOT = `${SPARTACUS_COMMERCE_QUOTES}/root`;
export const SPARTACUS_COMMERCE_QUOTES_ASSETS = `${SPARTACUS_COMMERCE_QUOTES}/assets`;
export const COMMERCE_QUOTES_TRANSLATIONS = 'commerceQuotesTranslations';
export const COMMERCE_QUOTES_TRANSLATION_CHUNKS_CONFIG =
  'commerceQuotesTranslationChunksConfig';
