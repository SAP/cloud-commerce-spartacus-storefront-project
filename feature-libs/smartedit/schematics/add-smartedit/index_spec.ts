import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  LibraryOptions as SpartacusSmartEditOptions,
  SpartacusOptions,
} from '@spartacus/schematics';
import * as path from 'path';
import { CLI_SMARTEDIT_FEATURE, SPARTACUS_SMARTEDIT } from '../constants';

const collectionPath = path.join(__dirname, '../collection.json');
const appModulePath = 'src/app/app.module.ts';

describe('Spartacus SmartEdit schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: any = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: any = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: 'scss',
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SpartacusSmartEditOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [CLI_SMARTEDIT_FEATURE],
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );
    schematicRunner.registerCollection(
      '@spartacus/organization',
      '../../feature-libs/organization/schematics/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('SmartEdit feature', () => {
    describe('assets', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultOptions, appTree)
          .toPromise();
      });

      it('should install @spartacus/smartedit library', () => {
        const packageJson = appTree.readContent('package.json');
        expect(packageJson).toContain(SPARTACUS_SMARTEDIT);
      });

      it('should add update angular.json with smartedit/assets', async () => {
        const content = appTree.readContent('/angular.json');
        const angularJson = JSON.parse(content);
        const buildAssets: any[] =
          angularJson.projects['schematics-test'].architect.build.options
            .assets;
        expect(buildAssets).toEqual([
          'src/favicon.ico',
          'src/assets',
          {
            glob: '**/*',
            input: './node_modules/@spartacus/smartedit/assets',
            output: 'assets/',
          },
        ]);

        const testAssets: any[] =
          angularJson.projects['schematics-test'].architect.test.options.assets;
        expect(testAssets).toEqual([
          'src/favicon.ico',
          'src/assets',
          {
            glob: '**/*',
            input: './node_modules/@spartacus/smartedit/assets',
            output: 'assets/',
          },
        ]);
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should add smartedit deps', async () => {
        const packageJson = appTree.readContent('/package.json');
        const packageObj = JSON.parse(packageJson);
        const depPackageList = Object.keys(packageObj.dependencies);
        expect(depPackageList.includes('@spartacus/smartedit')).toBe(true);
      });

      it('should import appropriate modules', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).toContain(
          `import { SmartEditRootModule } from '@spartacus/smartedit/root';`
        );
        expect(appModule).toContain(
          `import { SmartEditModule } from '@spartacus/smartedit';`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).not.toContain(`import('@spartacus/smartedit').then(`);
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultOptions, appTree)
          .toPromise();
      });

      it('should import SmartEditRootModule and contain the lazy loading syntax', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).toContain(
          `import { SmartEditRootModule } from '@spartacus/smartedit/root';`
        );
        expect(appModule).toContain(`import('@spartacus/smartedit').then(`);
      });

      it('should not contain the SmartEditModule import', () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).not.toContain(
          `import { SmartEditModule } from '@spartacus/smartedit';`
        );
      });
    });
  });

  describe('when other Spartacus features are already installed', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runExternalSchematicAsync(
          '@spartacus/organization',
          'ng-add',
          { ...spartacusDefaultOptions, name: 'schematics-test' },
          appTree
        )
        .toPromise();
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', defaultOptions, appTree)
        .toPromise();
    });

    it('should just append smartedit feature without duplicating the featureModules config', () => {
      const appModule = appTree.readContent(appModulePath);
      expect(appModule.match(/featureModules:/g)?.length).toEqual(1);
      expect(appModule).toContain(`smartEdit: {`);
    });
  });
});
