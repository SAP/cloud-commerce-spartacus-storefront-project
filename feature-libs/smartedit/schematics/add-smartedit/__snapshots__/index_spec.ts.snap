// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus SmartEdit schematics: ng-add SmartEdit feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { SmartEditModule } from \\"@spartacus/smartedit\\";
import { SmartEditRootModule } from \\"@spartacus/smartedit/root\\";

@NgModule({
  declarations: [],
  imports: [
    SmartEditRootModule,
    SmartEditModule
  ]
})
export class SmartEditFeatureModule { }
"
`;

exports[`Spartacus SmartEdit schematics: ng-add SmartEdit feature general setup assets should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\",
              {
                \\"glob\\": \\"**/*\\",
                \\"input\\": \\"./node_modules/@spartacus/smartedit/assets\\",
                \\"output\\": \\"assets/\\"
              }
            ],
            \\"styles\\": [
              \\"src/styles.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\",
              {
                \\"glob\\": \\"**/*\\",
                \\"input\\": \\"./node_modules/@spartacus/smartedit/assets\\",
                \\"output\\": \\"assets/\\"
              }
            ],
            \\"styles\\": [
              \\"src/styles.scss\\"
            ],
            \\"scripts\\": []
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:tslint\\",
          \\"options\\": {
            \\"tsConfig\\": [
              \\"tsconfig.app.json\\",
              \\"tsconfig.spec.json\\",
              \\"e2e/tsconfig.json\\"
            ],
            \\"exclude\\": [
              \\"**/node_modules/**\\"
            ]
          }
        },
        \\"e2e\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:protractor\\",
          \\"options\\": {
            \\"protractorConfig\\": \\"e2e/protractor.conf.js\\",
            \\"devServerTarget\\": \\"schematics-test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"schematics-test:serve:production\\"
            }
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus SmartEdit schematics: ng-add SmartEdit feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, provideConfig } from \\"@spartacus/core\\";
import { SmartEditRootModule, SMART_EDIT_FEATURE } from \\"@spartacus/smartedit/root\\";

@NgModule({
  declarations: [],
  imports: [
    SmartEditRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [SMART_EDIT_FEATURE]: {
        module: () =>
          import('@spartacus/smartedit').then((m) => m.SmartEditModule),
      },
    }
  })]
})
export class SmartEditFeatureModule { }
"
`;
